### PR TITLE
Atlendis integration

### PR DESCRIPTION
**Pull request by FredCoen**
## Description

Integrating the https://atlendis.io/. protocols position into zapper.

I am unsure about this implementation. Seeking guidance. Ideally I would like to have a look at the way Uniswap v3 positions have been implemented as a zapper integration cause the set up is similar to Atlendis positions. Each position is an nft that has distinct financial data attached to it. For example, amount lent out, amount not lent out and more. 

The current implementation gets all positions in atlendis within the position fetcher. This seems extremely inefficient. Also unsure how things are being relayed to the front end. Can it be tested how this is visualized in the frontend

Thanks for pointing us in the right direction

### CODE REVIEW

File: src/apps/atlendis-v1/polygon/atlendis-v1.balance-fetcher.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { ATLENDIS_V_1_DEFINITION } from '../atlendis-v1.definition';
+
+const network = Network.POLYGON_MAINNET;
+
+@Register.BalanceFetcher(ATLENDIS_V_1_DEFINITION.id, network)
+export class PolygonAtlendisV1BalanceFetcher implements BalanceFetcher {
+  constructor(@Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit) {}
+
+  async getPositionBalances(address: string) {
+    return this.appToolkit.helpers.contractPositionBalanceHelper.getContractPositionBalances({
```
Comment (immasandwich) [inline src/apps/atlendis-v1/polygon/atlendis-v1.balance-fetcher.ts:20]: Replace the body of `getPositionBalances` with the following logic:

1. Make a request to https://atlendis.herokuapp.com/graphql, requesting the positions as follows:

```gql
query {
  positions(address:"0x634ed8C2dF8fBe613eE25DEBE1a460B4DF2487d7") {
    tokenId
    normalizedAmount
    pool{
      parameters{
        underlyingToken
      }
    }
  }
}
```

(You'll find examples of the helper classes for making GQL queries here: https://studio.zapper.fi/docs/recipes/graphql-helper)

2. Once you have the token IDs, you can iterate over them, and build contract positions as:

```ts
const positions = data.positions.map(position => {
  const underlyingToken = baseTokens.find(
    t => t.address.toLowerCase() === position.pool.parameters.underlyingToken.toLowerCase(),
  );
  if (!underlyingToken) return null;

  const position = {
    type: ContractType.POSITION,
    appId,
    groupId,
    address: positionManagerAddress,
    network,
    tokens: [supplied(underlyingToken)],
    dataProps: {
      tokenId: position.tokenId,
      amount: position.normalizedAmount,
      owner: position.lender.address,
    },
    displayProps: {
      label: getLabelFromToken(underlyingToken),
      images: getImagesFromToken(underlyingToken),
    },
  };

  const tokenBalances = [drillBalance(position.tokens[0]), BigNumber.from(position.normalizedAmount).div(wad).mul(tokenDecimals).toString())]
  const contractPositionBalance: ContractPositionBalance = {
    ...position,
    tokens: tokenBalances,
    balanceUSD: sumBy(tokenBalances, v => v.balanceUSD),
  }

  return contractPositionBalance;
});
```

**Comment by FredCoen** [comment]
@immasandwich i think this is ready for review. as per my prior comment can you confirm this is how it will be rendered in the UI? The tokenId should be displayed instead of the tick range. the token balance should be displayed in the same way as above dollar value and units of token. Thanks

**Comment by immasandwich** [comment]
Looks great @FredCoen , can you please do one more favour and add the app logo to `src/atlendis/assets/logo.png`?

**Comment by SamIam-0x** [comment]
Hello @FredCoen !

Looks like you did not provide an address when submitting your PR. I recognize it may have seemed kind of random that we were asking for a wallet address with a PR submission, but we are actually looking for ways to reward Studio contributors in the future. 

Is there an address you’d want to provide? Could even be an empty address, if you do not have an doxxed address

Let me know! Also, if you’d like to privately provide it, feel free to shoot me a TG https://t.me/SamIAm_0x or DM on the Zapper Discord.

Thanks!
Sam

**Comment by FredCoen** [comment]
@SamIam-0x hey sure its 0x7453713a2539a9e9e25849eb8ed94c13e9e7138a

**Pull request merged**